### PR TITLE
feat: add storage_options to _BaseLanceDatasink, LanceDatasink, LanceCommitter

### PR DESCRIPTION
**Pull request by BitPhinix**
Adds the ability to pass `storage_options` to `_BaseLanceDatasink` to allow specifying them for `LanceDatasink` and `LanceCommitter` (`LanceFragmentWriter` already supports passing these).

We need to customize them because some of our datasets are hosted on `R2`

### CODE REVIEW

**Comment by BitPhinix** [comment]
Fixes https://github.com/lancedb/lance/issues/2615 as well

File: python/python/lance/ray/sink.py [hunk 1]
```diff
@@ -174,10 +183,18 @@ def __init__(
...
     ):
-        super().__init__(uri, schema=schema, mode=mode, *args, **kwargs)
+        super().__init__(
+            uri,
+            schema=schema,
+            mode=mode,
+            storage_options=storage_options,
+            *args,
+            **kwargs,
+        )
```
Comment (wjones127) [inline python/python/lance/ray/sink.py:198]: The `_write_fragment()` call below should also receive the `storage_options`, I think.

Comment (BitPhinix) [reply python/python/lance/ray/sink.py:198]: Done 👍

**Review by wjones127 — COMMENTED** [review]
Also, we validate storage options work in these tests: https://github.com/lancedb/lance/blob/main/python/python/tests/test_s3_ddb.py

Do want to add a test with the ray data sink there?

**Comment by wjones127** [comment]
Note: the CI failure is unrelated and will be fixed once this is #2620 merged. You should probably rebase on that is merged if you add the integration tests.

**Comment by BitPhinix** [comment]
Add some untested tests, not sure how to run integration tests locally to actually see if they work

File: python/python/tests/test_s3_ddb.py [hunk 2]
```diff
@@ -16,6 +16,8 @@
 import lance
 import pyarrow as pa
 import pytest
+from lance.dependencies import _RAY_AVAILABLE, ray
+from lance.ray.sink import LanceCommitter, LanceDatasink, LanceFragmentWriter
```
Comment (BitPhinix) [inline python/python/tests/test_s3_ddb.py:20]: Maybe I should move these imports into the tests?

**Comment by wjones127** [comment]
> Added some untested tests, not sure how to run integration tests locally to see if they work

Thanks. That's described in our development guide here:

https://github.com/lancedb/lance/blob/main/python/DEVELOPMENT.md#running-s3-integration-tests

**Review by wjones127 — APPROVED** [review]
Thank you @BitPhinix!

**Pull request merged**